libsolidity/interface/StandardCompiler.cpp

```diff
@@ -131,6 +132,11 @@ Json::Value formatErrorWithException(
 		formatSourceLocation(boost::get_error_info<errinfo_sourceLocation>(_exception)),
 		formatSecondarySourceLocation(boost::get_error_info<errinfo_secondarySourceLocation>(_exception))
 	);
+
+	if (_errorId.error != 0)
```

Comment:
Is this `if` mostly expected to be true or false?

Reply:
There are more places in code where the function is called with default `_errorId` parameter (0). But at runtime, under healthy circumstances, `_errorId` is supposed to be set (`if` would be `true`).

Do you suggest to invert the logic?

Reply:
Not necessarily, I just wanted to understand that's expected/exception